### PR TITLE
Fix crash when deleting measures from part score

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #17062 

(The commit titled "Refactoring:..." is not strictly necessary for this, but it's a related cleanup, that would be the completion of PR https://github.com/musescore/MuseScore/pull/11854.)

### CODE REVIEW

File: src/engraving/libmscore/score.h [hunk 1]
```diff
@@ -1233,8 +1229,9 @@ class Score : public EngravingObject
     virtual String name() const;
 
     void cmdTimeDelete();
-    void localTimeDelete();
-    void globalTimeDelete();
+    bool checkTimeDelete(Segment* startSegment, Segment* endSegment);
+    void doTimeDelete(Segment* startSegment, Segment* endSegment);
+    void doDoTimeDelete(Measure*, Segment*, const Fraction&);
```
Comment (RomanPudashkin) [inline src/engraving/libmscore/score.h:1234]: **doDo**TimeDelete? :) This name looks weird :)

**Review by RomanPudashkin — APPROVED** [review]

File: src/engraving/libmscore/score.h (hunk 1 above)
Comment (cbjeukendrup) [reply src/engraving/libmscore/score.h]: Yeah, I was a bit inspirationless 😅 Changed it to something a bit more meaningful now.

**Comment by DmitryArefiev** [comment]
Tested #17062 on Win10, Mac12, LinuxUbuntu 22.04 - **FIXED**

**Comment by DmitryArefiev** [comment]
Tested #17062 on Win10, Mac12, LinuxUbuntu 22.04 - **FIXED**

**Pull request merged**